frontend/src/Announcement.tsx

```diff
@@ -0,0 +1,80 @@
+import { Close } from '@mui/icons-material'
+import CampaignIcon from '@mui/icons-material/Campaign'
+import { Box, Button, Grid, IconButton, Stack, Typography } from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+import { useMemo, useState } from 'react'
+interface AnnoucementProps {
+  message: string
+  onClose: () => void
+}
+
+export default function Announcement({ message, onClose }: AnnoucementProps) {
+  const theme = useTheme()
+
+  const [showFullText, setShowFullText] = useState(false)
+
+  const announcementText = useMemo(() => {
+    return message.length > 100 ? (
+      <Stack>
+        <Typography>
+          {!showFullText ? `${message.slice(0, 100)}...` : message}
```

Comment:
Can we reverse the logic here? It's a little confusing to have a ternary written this way round. Consider:
```
{showFullText ? message : `${message.slice(0, 100)}...`}
```